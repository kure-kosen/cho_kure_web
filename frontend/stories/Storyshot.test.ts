import initStoryshots from "@storybook/addon-storyshots";
import {
  imageSnapshot,
  CustomConfig
} from "@storybook/addon-storyshots-puppeteer";

const storybookUrl = process.env.CIRCLECI
  ? "file:///home/circleci/repo/frontend/storybook-static"
  : "http://localhost:6006/";

const beforeScreenshot: CustomConfig["beforeScreenshot"] = (
  _page,
  _context
) => {
  return new Promise(resolve =>
    setTimeout(() => {
      resolve();
    }, 600)
  );
};

const getMatchOptions: CustomConfig["getMatchOptions"] = _context => ({
  // Allow 0.55% difference
  failureThreshold: 0.0055,
  failureThresholdType: "percent"
});

initStoryshots({
  suite: "Image storyshots",
  test: imageSnapshot({ storybookUrl, beforeScreenshot, getMatchOptions })
});
