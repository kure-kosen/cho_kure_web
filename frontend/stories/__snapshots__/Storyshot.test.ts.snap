// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots HeroArea Normal 1`] = `
<div
  className="sc-bdVaJa eoPsJS"
>
  HeroArea
</div>
`;

exports[`Storyshots HeroArea Personality Hero Area 1`] = `
<div
  className="sc-bdVaJa eoPsJS"
>
  <div
    className="sc-bwzfXH cmXBzf"
  >
    personality list
    <hr
      className="sc-htpNat jVvlcI"
    />
    過去にラジオに出演したメンバーをご紹介します
  </div>
</div>
`;
